app/models/batch_form.rb

```diff
@@ -67,7 +70,7 @@ def create_sample
       institution: self.institution,
       site: self.site,
       sample_identifiers: [SampleIdentifier.new],
-      date_produced: self.date_produced,
+      date_produced: @date_produced,
```

Comment:
This change shouldn't be necessary 🤔  since`date_produced` is a method declared in this class. 
I tried replacing it with the previous `self.date_produced` and something isn't working as before. 
I think we should find out what's the problem (given that the method itself had some implementation changes)

Reply:
`date_produced` is a field in `fields.yml` spcified as a `date`. Doing `self.date_produced` returns a string, so when we created a `Sample` with `self.date_produced` we were assigning a string type of value to the attribute. Trying `Sample.last.date_produced = Time.now` or `Sample.last.date_produced = '11/11/2021'` in a rails console results in a successful save operation, which means that the `Sample` model can be saved with either a `String` or a `Time` types. 
Also, a very importante detail: the date validation was being made in the `SampleForm` model, so when we were saving a `String` type to `date_produced` of the `Sample` model the validation was not being triggered. 

As of now, adding the validation to the `Sample` model triggers the validation and it fails since the value we are trying to save is a `String` type instead of a `Time` type. So we can't save `String` types to the `date_produced` field anymore. To comply with that I had two options: use `@date_produced` value or convert `self.date_produced` to Time (something like `Time.strptime(self.date_produced, Batch.date_format[:pattern])`).
Just to be cleaner, I used `@date_produced`, but any other suggestion is welcome!